zebra-network/src/protocol/external/codec.rs

```diff
@@ -113,6 +124,10 @@ impl Encoder for Codec {
             return Err(Parse("body length exceeded maximum size"));
         }
 
+        if self.builder.addr.is_some() {
+            metrics::counter!("bytes.written", (body.len() + HEADER_LEN) as u64, "addr" =>  self.builder.addr.unwrap().ip().to_string());
+        }
```

Comment:
the unwrap here can be avoided

```suggestion
        if let Some(addr) = self.builder.addr.as_ref() {
            metrics::counter!("bytes.written", (body.len() + HEADER_LEN) as u64, "addr" =>  addr.ip().to_string());
        }
```